subsys/net/openthread/Kconfig.defconfig

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# The purpose of this file is to define new default values of settings related
+# to building Thread from sources.
+# This file only changes defaults and thus all symbols here must be prompltless
+# and safeguarded so that they only are applied when building Thread.
+
+if NET_L2_OPENTHREAD
+
+choice LIBC_IMPLEMENTATION
+	default NEWLIB_LIBC
+endchoice
+
+# This is needed to properly handle MPSL in openthread samples
```

Comment:
and exactly why is this needed in OpenThread samples ?
https://github.com/nrfconnect/sdk-zephyr/blob/master/kernel/Kconfig#L100-L118
> 	int "Number of very-high priority 'preemptor' threads"

and
 
>This defines a set of priorities at the (numerically) lowest
> end of the range which have "meta-irq" behavior.  Runnable
> threads at these priorities will always be scheduled before
> threads at lower priorities, EVEN IF those threads are
> otherwise cooperative and/or have taken a scheduler lock.

What in openthread has this need ? and why is it needed ?
I think we can be a bit more informative to our users here.